examples/ecs/parallel_query.rs

```diff
@@ -0,0 +1,44 @@
+use bevy::{prelude::*, tasks::prelude::*};
+use std::{
+    sync::{atomic, atomic::AtomicUsize},
+    thread,
+    time::{Duration, Instant},
+};
+
+fn spawn_system(mut commands: Commands) {
+    for i in 0..16usize {
+        commands.spawn((i,));
+    }
+}
+
+fn square_system(pool: Res<ComputeTaskPool>, mut nums: Query<&mut usize>) {
+    let i = AtomicUsize::new(0);
+    nums.iter().iter_batched(1).for_each(&pool, |mut n| {
+        println!(
+            "Processing entity {}",
+            i.fetch_add(1, atomic::Ordering::Relaxed)
+        );
+        thread::sleep(Duration::from_secs(1));
```

Comment:
I like that this helps illustrate "expensive" tasks. This example helped prove to me that this works as expected, but I think I would prefer it if the example was a bit more practical.  I don't want newbies thinking that they need to throw thread::sleep or AtomicUsize in their parallel iterators.

Can we make this more like a "minimal example showing how to set up parallel execution"?